internal/schemas/builder.go

```diff
@@ -181,6 +197,48 @@ func (as *schemaImpl) BuildNodeFromList(lst []interface{}) (datamodel.Node, erro
 		case string:
 			lstItem := interface{}(val).(string)
 			la.AssembleValue().AssignString(lstItem)
+		case []byte:
+			lstItem := interface{}(val).([]byte)
+			la.AssembleValue().AssignBytes(lstItem)
+		/*
+			The below cases are for handling lists of up to 3 dimensions.
+			Within each cases arrays are normalized to match a type of []interface{}
+			each generic []interface{} array is then handed back to this function to further resolve types.
+			depth is cut off at 3 dimensions due to having to implement explicit type cases here
+		*/
+		case []string, []int, []int32, []int64, []float32, []float64:
+			value := make([]interface{}, 0)
+			s := reflect.ValueOf(val)
+			for i := 0; i < s.Len(); i++ {
+				value = append(value, s.Index(i).Interface())
+			}
+			n, err := as.BuildNodeFromList(value, kind.Item)
+			if err != nil {
+				return nil, err
+			}
+			la.AssembleValue().AssignNode(n)
+		case [][]byte, [][]string, [][]int, [][]int32, [][]int64, [][]float32, [][]float64:
+			value := make([]interface{}, 0)
+			s := reflect.ValueOf(val)
+			for i := 0; i < s.Len(); i++ {
+				value = append(value, s.Index(i).Interface())
+			}
+			n, err := as.BuildNodeFromList(value, kind.Item)
+			if err != nil {
+				return nil, err
+			}
+			la.AssembleValue().AssignNode(n)
+		case [][][]byte, [][][]string, [][][]int, [][][]int32, [][][]int64, [][][]float32, [][][]float64:
```

Comment:
Would generics work here?

Reply:
This will be updated with the new schema defs